src/services/review/RepoCheckerService.ts

```diff
@@ -4,110 +4,358 @@ import path from "path"
+    this.gitFileSystemService = gitFileSystemService
+    this.git = git
+    this.repoRepository = repoRepository
   }
 
-  runCheckerForAllRepos(): ResultAsync<undefined, Error> {
-    // todo
-    return errAsync(new Error("Not implemented"))
+  /**
+   * NOTE: This gets all the repos that have site members only
+   * We want to exclude GH-login folks from using this feature
+   */
+  getAllRepos(): ResultAsync<string[], SiteCheckerError> {
+    return ResultAsync.fromPromise(
+      this.siteMemberRepository.findAll({
+        include: [
+          {
+            model: Site,
+            required: true,
+            include: [
+              {
+                model: Repo,
+                required: true,
+              },
+              {
+                model: Deployment,
+                required: true,
+              },
+            ],
+          },
+        ],
+      }),
+      (error) => new SiteCheckerError(`${error}`)
+    )
+      .map((a) => a.map((b) => b.dataValues.site.repo.dataValues.name))
+      .andThen((repos) => {
+        const uniqueRepos: string[] = Array.from(
+          new Set(repos.filter((repo) => repo))
+        )
+        return ok(uniqueRepos)
+      })
   }
 
-  async checkRepo(repo: string) {
+  getStagingLinkFromRepoName(
+    repoName: string
+  ): ResultAsync<string, SiteCheckerError> {
+    return ResultAsync.fromPromise(
+      this.repoRepository.findOne({
+        where: {
+          name: repoName,
+        },
+        include: [
+          {
+            model: Site,
+            required: true,
+            include: [
+              {
+                model: Deployment,
+                required: true,
+              },
+            ],
+          },
+        ],
+      }),
+      (error) => new SiteCheckerError(`${error}`)
+    ).andThen((repo) => {
+      // we still return something even if not found as
+      // the permalink might still be useful to user
+      if (!repo) return ok("")
+      return okAsync(repo.dataValues.site.deployment.dataValues.stagingUrl)
+    })
+  }
+
+  runCheckerForAllRepos(): ResultAsync<
+    undefined,
+    SiteCheckerError | SiteCheckerError[]
+  > {
+    return this.getAllRepos().andThen((repos) => this.runCheckerForRepos(repos))
+  }
+
+  runCheckerForRepos(
+    repos: string[]
+  ): ResultAsync<undefined, SiteCheckerError[]> {
+    return ResultAsync.fromPromise(
+      this.runCheckerForReposInSeq(repos),
+      (error) => [new SiteCheckerError(`${error}`)]
+    )
+      .andThen((result) => ok(undefined))
+      .orElse((errors) => {
+        logger.error(`Error running site checker for repos: ${errors}`)
+        return errAsync(errors)
+      })
+  }
+
+  /**
+   * Run the site checker for all repos sequentially
+   * Not using neverthrow here as we want to run all the checkers
+   * in sequence without short circuiting, and neverthrow does not
+   * support this.
+   */
+  async runCheckerForReposInSeq(repos: string[]): Promise<void> {
```

Comment:
the comment seems to suggest that we want this to be the case?

Reply:
@seaerchin ps typo, this ran concurrently** while we want sequentially

Reply:
this is not important now, but why prefer sequential > concurrent if it doesn't short circuit? 

this would only be the case if we hold some lock or our current operation depends on a prior operation, neither of which seems to be the case

Reply:
hey i dont have a strong argument apart from fear of using compute resources. ie i dont want to accidentally ddos us by running a program to parse 600+ folders at once. 

but actually this seems like a pretty trivial worry since node should handle this for us + io ops have pretty generous limits. 

i am prioritising pushing the current working prototype to prod first + and then rethink this/see if can use your gh actions idea

Reply:
I think we would add a gh action for every repo to check on push to staging and pr to masterz so this wouldn't be an issue.

There wouldn't be a ddos concern because this should be a script or docker container that we can just pull and use as is off of github's resources

Reply:
i do intend to try this but gh actions are known to take a while and we have a realtime requirement of 1 minute to meet, can try later and see if this works as intended 

also, there is a push to reduce dep on gh, so i am thinking of timeboxing this effort too, thoughts?